### DISCUSSION TITLE
Map iteration crashes client

### DISCUSSION THREAD

**Discussion by miguelalbuquerque** [scandum/tintin#79] [Bugs]
Hi,
My map has above 60k rooms.
I'm trying to develop some scripts which need to iterate through all the rooms, but getting a client crash. 

To demonstrate, I can do something like the following:
#map read <file>
#map goto
#map list {variable} {maplist}
#foreach {*maplist[]}{temp} {#nop do nothing for now}

The result:

DEBUG_STACK[000] [000] = mainloop()
DEBUG_STACK[001] [000] = process_input(void)
DEBUG_STACK[002] [001] = script_driver(%p,%d,%p)
DEBUG_STACK[003] [001] = sub_arg_in_braces(%p,%p,%p,%d,%d)
DEBUG_STACK[004] [002] = substitute(%p,%p,%p,%d)
DEBUG_STACK[005] [005] = substitute(%p,%p,%p,%d)

FATAL ERROR trap_handler (signal 11: Segmentation fault)
syserr_fatal(crashed)

If I change the index to be restricted to, for example only 1000 rooms, there is no crash.

Is this due to the size of the map?
I've tried in my desktop (wintin) and in iSH (ipad), same result.

Thanks for any hint.

**Comment by scandum** [comment]
#foreach causes a buffer overflow if the variable is too large. Best solution is to use #loop instead.

```
#map list {variable} {maplist}
#loop 1 &maplist[] index {#showme vnum: *maplist[+$index]}
```

**Reply by miguelalbuquerque** [reply]
Thank you @scandum ! Will try it indeed.

**Reply by miguelalbuquerque** [reply]
Just to confirm that it worked! Thanks again!
